Extractor/Extractor.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace Bazinga\Bundle\JsTranslationBundle\Extractor;
+
+use Symfony\Component\Translation\Extractor\AbstractFileExtractor;
+use Symfony\Component\Translation\MessageCatalogue;
+use Symfony\Component\Translation\Extractor\ExtractorInterface;
+use Bazinga\Bundle\JsTranslationBundle\Finder\FinderFactory;
+use Bazinga\Bundle\JsTranslationBundle\Filesystem\Filesystem;
+
+abstract class Extractor extends AbstractFileExtractor implements ExtractorInterface
+{
+    const FUNCTION_STRING_ARGUMENT_REGEX = '\s?[\'"]([^"\'),]+)[\'"]\s?';
+    const REGEX_DELIMITER = '/';
+
+    private $prefix = '';
+
+    private $filesystem;
+
+    private $finderFactory;
+
+    public function __construct(Filesystem $filesystem, FinderFactory $finderFactory) {
+        $this->filesystem = $filesystem;
+        $this->finderFactory = $finderFactory;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function extract($resource, MessageCatalogue $catalogue)
+    {
+        $assetsPath = dirname($resource) . '/public';
+
+        if (!$this->filesystem->exists($assetsPath)) {
+            return;
+        }
+
+        $files = $this->extractFiles($assetsPath);
+
+        foreach ($files as $file) {
+            $this->parseMessagesFromContent($this->filesystem->getContents($file), $catalogue);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setPrefix($prefix)
+    {
+        $this->prefix = $prefix;
+    }
+
+    protected function parseMessagesFromContent($fileContent, MessageCatalogue $catalogue)
```

Comment:
all these should be private. There is no reason to introduce extension points needing to be maintained for BC.